solr/modules/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java

```diff
@@ -241,10 +243,28 @@ public void setContext(ResultContext context) {
      *
      * @param transformerFeatureStore the explicit transformer feature store
      */
-    private LoggingModel createLoggingModel(String transformerFeatureStore) {
-      final ManagedFeatureStore fr = ManagedFeatureStore.getManagedFeatureStore(req.getCore());
+    private LoggingModel createLoggingModel(
+        String transformerFeatureStore, Boolean docsWereNotReranked) {
+      final ManagedFeatureStore featureStore =
+          ManagedFeatureStore.getManagedFeatureStore(req.getCore());
+
+      // check for non-existent feature store name
+      if (transformerFeatureStore != null) {
+        if (!featureStore.getStores().containsKey(transformerFeatureStore)) {
+          throw new FeatureException("Missing feature store: " + transformerFeatureStore);
+        }
+      }
+
+      final FeatureStore store = featureStore.getFeatureStore(transformerFeatureStore);
 
-      final FeatureStore store = fr.getFeatureStore(transformerFeatureStore);
+      // check for empty feature store only if there is no reranking query, otherwise the model
+      // store will be used later for feature extraction
+      if (docsWereNotReranked) {
+        if (store.getFeatures().isEmpty()) {
+          throw new FeatureException(
+              "Empty feature store. If no ranking query (rq) and DEFAULT feature store is empty, an existing feature store name must be passed for feature extraction.");
```

Comment:
"Empty feature store. If no ranking query (rq) is passed and DEFAULT feature store is empty, an existing feature store name must be passed for feature extraction."